### PR TITLE
Add JSON functionality to SQLAlchemy Dialect

### PR DESCRIPTION
**Pull request by laserkaplan**
## Description

Implement the ability to reference JSON columns using SQLAlchemy. This allows JSON columns to be created in new tables, as well as writes to/reads from them using normal Python `json` module serialization/deserialization.

Originally referenced [in this issue](https://github.com/trinodb/trino-python-client/issues/194).

## Release notes

( ) This is not user-visible and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
* Add support for creating tables containing `JSON` columns and reading and writing to them with SQLAlchemy. ([#194](https://trinodb/trino-python-client/issues/194))
```

### CODE REVIEW

File: trino/sqlalchemy/dialect.py
Comment (hovaesco) [inline trino/sqlalchemy/dialect.py:72]: Based on the PR description:
> using normal Python json module serialization/deserialization

deserialization should be also done using `json` module, why then here it's `None`?

Comment (laserkaplan) [reply trino/sqlalchemy/dialect.py:72]: Yes, good eye :)
This is actually because of a minor bug in the SQLAlchemy code. See here: https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/sql/sqltypes.py#L2627
If you don't specify a `_json_serializer` or `_json_deserializer` at all in the dialect, then this will throw an error trying to access an attribute that doesn't exist.
However, if you try to set this attribute to just `json.loads`, Python thinks it's an instance method, and expects 2 parameters to be passed (one of which is `self`), and thus causes the downstream call to fail.
I can get around this by setting `_json_deserializer = staticmethod(json.loads)` if you think this is more clear. I can also just add a comment to this effect.

File: trino/sqlalchemy/dialect.py
Comment (laserkaplan) [reply trino/sqlalchemy/dialect.py:67]: Thanks! Great suggestion, I will absolutely look into this and update the PR.
SQLAlchemy is such a massive package that it's easy to miss good functionality like this :)

File: trino/sqlalchemy/types.py
Comment (hovaesco) [inline trino/sqlalchemy/types.py:17]: This should go to `trino/sqlalchemy/datatype.py`

**Review by mdesmet — APPROVED** [review]
@laserkaplan: please have a look at the pending remark of @hovaesco on moving the TypeDecorator to datatype.py.

LGTM % the above remark and squash the commits.

**Comment by laserkaplan** [comment]
Whoops, thanks for the reminder! Moved to `datatype.py`.

**Comment by laserkaplan** [comment]
Seems like these tests failing might be a larger issue? Multiple open PRs are all failing at the same spot.

**Comment by mdesmet** [comment]
> Seems like these tests failing might be a larger issue? Multiple open PRs are all failing at the same spot.

Yes, it should be gone when #220 is merged.

**Review by hovaesco — APPROVED** [review]
Please squash commits.

**Review by hashhar — APPROVED** [review]
I'll merge this after I merge https://github.com/trinodb/trino-python-client/pull/220 tomorrow morning my time.

Thanks for contributing @laserkaplan and sorry about the very long delay.

**Pull request merged**
